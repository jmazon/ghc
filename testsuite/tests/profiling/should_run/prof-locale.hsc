{-# LANGUAGE ForeignFunctionInterface #-}

import Control.Exception
import Foreign.C.Types
import Foreign.C.String

#include <locale.h>
lcAll = #const LC_ALL
foreign import ccall "locale.h setlocale" c_setlocale :: CInt -> CString -> IO ()
foreign import ccall  "stdio.h printf"    c_printf :: CString -> CDouble -> IO ()

main = do
  withCString "" $ c_setlocale lcAll

  -- allocate some junk to get a heap profile
  s <- getLine
  evaluate $ last $ scanl1 (+) $ map fromEnum $ concat $ replicate 100000 s
  -- check 0 (test validity): ensure we got a profile (sed|grep in Makefile)
      
  -- check 1: RTS doesn't override user's locale setting
  withCString "%'.3f\n" $ \fmt -> c_printf fmt (read s)

  -- check 2: heap profile is readable by hp2ps (hp2ps in Makefile)
