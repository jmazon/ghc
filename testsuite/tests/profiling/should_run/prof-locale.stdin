3141.592653589793
