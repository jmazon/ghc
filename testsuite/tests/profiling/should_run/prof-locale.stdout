[1 of 1] Compiling Main             ( prof-locale.hs, prof-locale.o )
Linking prof-locale ...
3â€¯141,593
